geopandas/geodataframe.py

```diff
@@ -459,6 +459,17 @@ def __getitem__(self, key):
             result.__class__ = DataFrame
         return result
 
+
+    def explode(self):
+        """
+        Overriden from GeoPandasBase, returns rest of the dataframe joined to 
+        exploded geometry column.
+        """
+        exploded = self._get_geometry().explode().reset_index(level=1, drop=True)
+        geo_df = self.drop(columns='geometry').join(exploded)
```

Comment:
Here we should not hardcode the 'geometry' name, as it can be something else. You can use `self._geometry_column_name` instead

Reply:
Fixed.